sdk/tables/data-tables/src/serialization.ts

```diff
@@ -157,23 +157,35 @@ export function deserialize<T extends object = Record<string, any>>(
   return deserialized;
 }
 
-function inferTypedObject(propertyName: string, value: number | string) {
+function inferTypedObject(propertyName: string, value: number | string | boolean) {
   // We need to skip service metadata fields such as partitionKey and rowKey and use the same value returned by the service
   if (propertyCaseMap.has(propertyName)) {
     return value;
   }
 
-  return typeof value === "string" ? { value, type: "String" } : getTypedNumber(value);
+  const valuStr = String(value);
+
+  switch (typeof value) {
+    case "boolean":
+      return { value: valuStr, type: "Boolean" };
+    case "number":
+      return getTypedNumber(value);
+    case "string":
+      return { value: valuStr, type: "String" };
+    default:
+      return value;
+  }
 }
 
 /**
  * Returns the number when typeConversion is enabled or the EDM object with the correct number format Double or Int32 if disabled
  */
-function getTypedNumber(value: number): EdmModel<"Double"> | EdmModel<"Int32"> {
+function getTypedNumber(value: number): { value: string; type: "Int32" | "Double" } {
+  const valueStr = String(value);
```

Comment:
technically this will lose precision for some cases. Does the server return it to us as a JSON number instead of as a string so there's not much we can do?